gunicorn/workers/geventlet.py

```diff
@@ -125,6 +130,10 @@ def patch(self):
         patch_sendfile()
 
     def is_already_handled(self, respiter):
+        # eventlet >= 0.30.3
+        if EVENTLET_WSGI_LOCAL and EVENTLET_WSGI_LOCAL.already_handled:
```

Comment:
why do we need an « and » there ? can’t we simply check if one or the other is present and set to true?

Reply:
Sorry, I don't understand.

We need first check because there may be older version of eventlet and module level getattr would set it to None. Arguably should explicitly compare to None like so.

`if EVENTLET_WSGI_LOCAL is not None:`  # eventlet >= 0.30.3

And then if we know it's eventlet with WSGI_LOCAL, we can check its `already_handled` and avoid resolving `None.already_handled`.

Surely you got all this and I'm just confused about what other option is possible here. Sorry. If you show a more concrete example, I will change code as you wish.

Reply:
I think another way of expressing this would be to write `if hasattr(EVENTLET_WSGI_LOCAL, 'already_handled')`.

I am not saying this is better, I am just stating @temoto's case another way, to help illustrate the intent. (Or at least, that's how I undersatnd it.)

Reply:
@benoitc did you want getattr like that?

Reply:
Changed to single `getattr`, it would work in earlier versions of eventlet.